src/test/java/org/openmrs/addonindex/VersionServiceIT.java

```diff
@@ -0,0 +1,56 @@
+package org.openmrs.addonindex;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.openmrs.addonindex.domain.AddOnInfoAndVersions;
+import org.openmrs.addonindex.domain.AddOnType;
+import org.openmrs.addonindex.domain.AddOnVersion;
+import org.openmrs.addonindex.service.Index;
+import org.openmrs.addonindex.service.artifactory.VersionsService;
+import org.openmrs.addonindex.util.Version;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest
+@RunWith(SpringRunner.class)
+public class VersionServiceIT
```

Comment:
You have structured this IT correctly. Good work!

However it doesn't make sense to actually be integration-testing this as part of the service since all of the logic is carried out by the domain object. It's sufficient to unit test this in the add on class. (We can also have an end-to-end test in the controller, but we don't need this tested in all three places.)